src/libData/AccountData/Transaction.cpp

```diff
@@ -227,7 +228,14 @@ const uint128_t Transaction::GetGasPriceWei() const {
   }
 }
 
-const uint64_t& Transaction::GetGasLimit() const { return m_coreInfo.gasLimit; }
+uint64_t Transaction::GetGasLimit() const {
+  if (IsEth()) {
+    return GasConv::GasUnitsFromEthToCore(m_coreInfo.gasLimit);
+  }
+  return m_coreInfo.gasLimit;
+}
+
+uint64_t Transaction::GetGasLimitRaw() const { return m_coreInfo.gasLimit; }
```

Comment:
It would be helpful if you defined `GetGasLimitEth` and `GetGasLimitZil` for readability.  It is hard to understand.  Why would I want to use 'Raw' for Eth, for instance?

Then you may need to convert in the other direction.  If it is an non-Eth transaction, but is somehow used in an Eth calculation context, you'll scale it too.  (There's a use case for that in the future).

Then the only place where you use raw is in signatures.  OTherwise you always use either *Zil or *Eth, and there's no need to guess.